samples/bluetooth/mesh/chat/README.rst

```diff
@@ -0,0 +1,333 @@
+.. _bluetooth_mesh_chat:
+
+Bluetooth: Mesh Chat
+####################
+
+The Bluetooth Mesh Chat sample demonstrates how the mesh network can be used to facilitate communication between nodes by text, using the :ref:`bluetooth_mesh_chat_client_model`.
+By means of the mesh network, the clients as mesh nodes can communicate with each other without the need of a server.
+The sample is mainly designed for group communication, but it also supports one-on-one communication, as well as sharing the nodes presence.
+
+This sample is used in :ref:`ug_bt_mesh_vendor_model` as an example of how to implement a vendor model for the Bluetooth Mesh in |NCS|.
+
+Overview
+********
+
+This sample is split into four source files:
+
+* A :file:`main.c` file to handle initialization.
+* A file for handling the Chat Client model, :file:`chat_cli.c`.
+* A file for handling Bluetooth Mesh models, :file:`model_handler.c`.
+* A file for handling communication with UART, :file:`uart_handler.c`.
+
+After provisioning and configuring the Bluetooth Mesh models supported by the sample in the `nRF Mesh mobile app`_, you can communicate with other mesh nodes by sending text messages and obtaining their presence.
+
+Provisioning
+============
+
+The provisioning is handled by the :ref:`bt_mesh_dk_prov`.
+
+Models
+======
+
+The following table shows the Bluetooth Mesh Chat composition data for this sample:
+
+   +---------------+
+   |  Element 1    |
+   +===============+
+   | Config Server |
+   +---------------+
+   | Health Server |
+   +---------------+
+   | Chat Client   |
+   +---------------+
+
+The models are used for the following purposes:
+
+* The :ref:`bluetooth_mesh_chat_client_model` instance in the first element is used to communicate with the other Chat Client models instantiated on the other mesh nodes.
+* Config Server allows configurator devices to configure the node remotely.
+* Health Server provides ``attention`` callbacks that are used during provisioning to call your attention to the device.
+  These callbacks trigger blinking of the LEDs.
+
+The model handling is implemented in :file:`src/model_handler.c`.
+
+.. _bluetooth_mesh_chat_client_model:
+
+Chat Client model
+-----------------
```

Comment:
I think this should be split out to a separate subpage that follows the model template. It's mostly this first paragraph and the states and messages that fits there though.